### ISSUE TITLE
When we click "Save text as finished" twice, user_state will be wrong

### ISSUE THREAD

**Issue by aufziehvogel** [aufziehvogel/sprakit-website#7]
E.g.: You currently have 402 text length score. Now you read a simple text and should drop by two, so when you click on "Save text as finished" you drop to 400. Now you make some adjustments to the vocabulary (but not much) and should drop about 2 still (so 400 would still be right). The adjustment will be done to 400 once again, so your new score is 398...

There are also multiple dots in the user score chart then, this should also not happen, it should not display one dot per text.

In short: clicking on a "Save text as finished" for the second time should trigger a correct edit, not a total new finished procedure.
